src/Api/Controllers/CollectionsController.cs

```diff
@@ -397,27 +404,17 @@ public async Task DeleteUser(Guid orgId, Guid id, Guid orgUserId)
         await _collectionService.DeleteUserAsync(collection, orgUserId);
     }
 
-    private void DeprecatedPermissionsGuard()
-    {
-        if (FlexibleCollectionsIsEnabled)
-        {
-            throw new FeatureUnavailableException("Flexible Collections is ON when it should be OFF.");
-        }
-    }
```

Comment:
I propose removing this for a couple of reasons
1. it was probably overkill - we ended up separating the current endpoints and the `vNext` endpoints pretty cleanly, and this controller has reasonable test coverage, so we have some confidence that the correct code paths are being run
2. now that we're doing our data migration earlier, there's less risk in the wrong authorization code being run. Previously, the manager enum was still in the database, so the wrong code path might authorize an unauthorized action. Now managers really will be downgraded to users, even if the old authorization logic is accidentally called, the worst case is that they'll get rejected - which is what this guard did anyway.

Therefore I didn't worry about calling the orgAbilities cache and making all this async - I just removed it. Open for discussion though, it's not a huge lift to restore it.